### PR TITLE
Google Sheets Destination

### PR DESCRIPTION
**Pull request by utsabc**
## Google Sheets destination

> Google sheets destination

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/googlesheets/transform.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const {
+  getHashFromArray,
+  getMappingConfig,
+  constructPayload
+} = require("../../util");
+
+const CONFIG_CATEGORIES = {
```
Comment (ruchiramoitra) [inline v0/destinations/googlesheets/transform.js:7]: take these to config.js similar to other destinations.

File: v0/destinations/googlesheets/transform.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const process = event => {
+  const { message, destination } = event;
+  const tabName = getTabName(event);
+  if (tabName) {
+    const payload = constructPayload(
+      message,
+      MAPPING_CONFIG.GoogleSheetsMapping
+    );
+    payload.spreadSheetId = destination.Config.sheetId;
+    payload.spreadSheetTab = tabName;
```
Comment (ruchiramoitra) [inline v0/destinations/googlesheets/transform.js:33]: send payload.userId too.

File: v0/destinations/googlesheets/data/GoogleSheetsMapping.json [hunk 3]
```diff
@@ -0,0 +1,172 @@
+[
+  {
+    "destKey": "anonymous_id",
```
Comment (ruchiramoitra) [inline v0/destinations/googlesheets/data/GoogleSheetsMapping.json:3]: also add user_id

File: v0/destinations/googlesheets/data/GoogleSheetsMapping.json [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    "sourceKeys": "context.userAgent",
+    "required": false
+  },
+  {
+    "destKey": "event",
+    "sourceKeys": "event",
+    "required": false
+  },
+  {
+    "destKey": "event_name",
```
Comment (ruchiramoitra) [inline v0/destinations/googlesheets/data/GoogleSheetsMapping.json:138]: not needed

File: v0/destinations/googlesheets/data/GoogleSheetsMapping.json [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    "sourceKeys": ["event", "name"],
+    "required": false
+  },
+  {
+    "destKey": "id",
+    "sourceKeys": "messageId",
+    "required": false
+  },
+  {
+    "destKey": "original_timestamp",
```
Comment (ruchiramoitra) [inline v0/destinations/googlesheets/data/GoogleSheetsMapping.json:153]: use timestamp from generic mapping

File: v0/destinations/googlesheets/data/GoogleSheetsMapping.json [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    "required": false
+  },
+  {
+    "destKey": "received_at",
+    "sourceKeys": "receivedAt",
+    "required": false
+  },
+  {
+    "destKey": "timestamp",
+    "sourceKeys": "timestamp",
```
Comment (ruchiramoitra) [inline v0/destinations/googlesheets/data/GoogleSheetsMapping.json:164]: use historical timestamp from generic mapping

File: v0/destinations/googlesheets/transform.js [hunk 7]
```diff
@@ -9,6 +9,7 @@ const getTabName = event => {
 
   return (
     (message.event ? hashMap[message.event.toLowerCase()] : null) ||
+    (message.name ? hashMap[message.name.toLowerCase()] : null) ||
```
Comment (utsabc) [inline v0/destinations/googlesheets/transform.js:12]: Added support for `message.name` for mapping to Tab in sheets (WDYT @ruchiramoitra )
Example event like
page call with a name `Home Page` to be mapped to a specific tab so in scenarios where `event` property is not present but `name` we will consider `name` for mapping the event to a Specific Sheet tab. Thoughts?

**Review by ruchiramoitra — APPROVED** [review]

**Review by ruchiramoitra — APPROVED** [review]

**Review by ruchiramoitra — APPROVED** [review]

**Review by ruchiramoitra — APPROVED** [review]

**Pull request merged**